### ISSUE TITLE
Fix false negatives for CSS-In-JS in color-no-invalid-hex

### ISSUE THREAD

**Issue by alflennik** [stylelint/stylelint#3945]
Stylelint does not catch CSS errors in JSX, and according to https://github.com/stylelint/stylelint/issues/3264, out-of-the-box JSX support is expected.

```javascript
// Test.jsx
import React from "react";

export default function Test() {
  return <h1 style={{ color: "#FFFFF" }}>Test</h1>; // Invalid hex code
}
```

```javascript
// .stylelintrc
{
  "extends": ["stylelint-config-recommended"]
}
```

```javascript
// package.json
{
  "scripts": {
    "stylelint": "stylelint"
  },
  "dependencies": {
    "react": "16.8.0"
  },
  "devDependencies": {
    "stylelint": "^9.10.1",
    "stylelint-config-standard": "^18.2.0",
  }
}
```

Then, running `yarn stylelint Test.jsx` has the following output:

```
$ stylelint Test.jsx
✨  Done in 0.84s.
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

It is related to JSX syntax.

> What did you expect to happen?

It should lint the style prop.

> What actually happened (e.g. what warnings or errors did you get)?

No linting errors.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

I can reproduce it in a demo.

It should be linted, and postcss-jsx (a package we use for processing CSS-in-JS) is [supported](https://github.com/gucong3000/postcss-jsx/blob/master/test/react.js) this type of CSS-in-JS.

**Comment by jeddy3** [comment]
@alflennik Thanks for the report.

This seems to be specific to the `color-no-invalid-hex` rule.

The following:

```js
// Test.jsx
import React from "react";

export default function Test() {
  return <h1 style={{ unknown: "#FFFF"}}>Test</h1>; // Unknown property
}
```

Produces the following violation:

```
5:22 error Unexpected unknown property "unknown" (property-no-unknown)
```

I suspect it's because of the rule's use of `toString` on the `decl` node or the use of `styleSearch`:

https://github.com/stylelint/stylelint/blob/0f5474d850380b3c4948fa5f26c052a135dcdeac/lib/rules/color-no-invalid-hex/index.js#L24-L26

It'll hopefully be a quick fix, though.

If you'd like to contribute a fix, there are steps in the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) on how to get start.
